### ISSUE TITLE
Fix CORS headers for image proxying

### ISSUE THREAD

**Issue by beaugunderson** [LockerProject/Locker#787]
This is related to #766.

As you can [see here](https://gist.github.com/1793921), we're currently sending two copies of 'Access-Control-Allow-Origin'.

I'm not sure if that's the entire issue, but CORS requests for images that include proxy=1 are still being forbidden by the browser in both Chrome and Firefox. I'll follow this up with some additional testing results.

**Comment by beaugunderson** [comment]
From talking to @othiym23 it appears that the Singly proxy is adding its own Access-Control-Allow-Origin CORS header. It appears that the logic that adds that header should be modified to only do so if an existing Access-Control-Allow-Origin header isn't set.

I verified that cross-origin image loading works with a simple Express setup that mirrors how I was using it via the Singly API (namely adding an img to the DOM with jQuery, then setting its crossOrigin and src attributes).

**Comment by beaugunderson** [comment]
This was working for a bit and then stopped--I think the proxy still needs looking at.

**Comment by beaugunderson** [comment]
More information: The `crossOrigin` img attribute strips the HTTP request of cookies--meaning there's no `singly.id` session cookie sent.

When a request is sent for `/Me/photos/thumbnail/:id?proxy=1` with no cookies a 302 redirect is returned to https://singly.com.

cc @quartzjer

**Comment by beaugunderson** [comment]
Workaround: Instead of using `me.singly.com` you can use `api.singly.com/:apiToken/` (because it does't require a session cookie).

**Comment by quartzjer** [comment]
Nice sleuthing and workaround! Is it even possible to make it work the normal way?

**Comment by beaugunderson** [comment]
I had a small epiphany last night. I started out down the CORS path because I wanted to be able to develop locally (either with file:// or with my own webserver). On me.singly.com I don't think it will actually even need to use CORS since the domains will match! The `proxy=1` parameter will be enough to make it work on the site. For developing locally I found out about the ominous-sounding (but extremely useful) `--disable-web-security` flag in Chrome. I'm not sure if I have to go through the extra step of making the img requests use CORS if I use that flag, but I'll do some testing and report back. Once I've figured out what works I'll write up a wiki page to document everything. :)
